{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Pathogenicity</th>\n",
       "      <th>indel</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Protein.stable.ID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Func</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD_popmax</th>\n",
       "      <th>gnomAD3</th>\n",
       "      <th>UKBB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:97611611:TGAG:T</td>\n",
       "      <td>10</td>\n",
       "      <td>97611611</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000241935</td>\n",
       "      <td>ENST00000370646</td>\n",
       "      <td>ENSP00000359680</td>\n",
       "      <td>HOGA1</td>\n",
       "      <td>2</td>\n",
       "      <td>313</td>\n",
       "      <td>E/-</td>\n",
       "      <td>0.000465753</td>\n",
       "      <td>0.000123235</td>\n",
       "      <td>0.000269343</td>\n",
       "      <td>9.71885e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:124405531:TAGC:T</td>\n",
       "      <td>10</td>\n",
       "      <td>124405531</td>\n",
       "      <td>TAGC</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000065154</td>\n",
       "      <td>ENST00000368845</td>\n",
       "      <td>ENSP00000357838</td>\n",
       "      <td>OAT</td>\n",
       "      <td>2</td>\n",
       "      <td>184</td>\n",
       "      <td>A/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19:33387464:GTCC:G</td>\n",
       "      <td>19</td>\n",
       "      <td>33387464</td>\n",
       "      <td>GTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000124299</td>\n",
       "      <td>ENST00000244137</td>\n",
       "      <td>ENSP00000244137</td>\n",
       "      <td>PEPD</td>\n",
       "      <td>2</td>\n",
       "      <td>453-454</td>\n",
       "      <td>ED/D</td>\n",
       "      <td>8.04531e-06</td>\n",
       "      <td>8.88952e-06</td>\n",
       "      <td>1.31416e-05</td>\n",
       "      <td>7.47604e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22:42619859:ACAT:A</td>\n",
       "      <td>22</td>\n",
       "      <td>42619859</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>M/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22:42619781:CGAA:C</td>\n",
       "      <td>22</td>\n",
       "      <td>42619781</td>\n",
       "      <td>CGAA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>299</td>\n",
       "      <td>F/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5452</th>\n",
       "      <td>21:39779227:C:CGCTGCT</td>\n",
       "      <td>21</td>\n",
       "      <td>39779227</td>\n",
       "      <td>C</td>\n",
       "      <td>CGCTGCT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000183067</td>\n",
       "      <td>ENST00000380588</td>\n",
       "      <td>ENSP00000369962</td>\n",
       "      <td>IGSF5</td>\n",
       "      <td>2</td>\n",
       "      <td>286</td>\n",
       "      <td>R/RCC</td>\n",
       "      <td>0.00241128</td>\n",
       "      <td>0.00413725</td>\n",
       "      <td>0.00279994</td>\n",
       "      <td>0.00421913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5453</th>\n",
       "      <td>22:20566526:A:ACAGCAGCAG</td>\n",
       "      <td>22</td>\n",
       "      <td>20566526</td>\n",
       "      <td>A</td>\n",
       "      <td>ACAGCAGCAG</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000099917</td>\n",
       "      <td>ENST00000263205</td>\n",
       "      <td>ENSP00000263205</td>\n",
       "      <td>MED15</td>\n",
       "      <td>2</td>\n",
       "      <td>250-251</td>\n",
       "      <td>-/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0022583</td>\n",
       "      <td>0.00060243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5454</th>\n",
       "      <td>22:27798945:T:TTGCTGC</td>\n",
       "      <td>22</td>\n",
       "      <td>27798945</td>\n",
       "      <td>T</td>\n",
       "      <td>TTGCTGC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000169184</td>\n",
       "      <td>ENST00000302326</td>\n",
       "      <td>ENSP00000304956</td>\n",
       "      <td>MN1</td>\n",
       "      <td>2</td>\n",
       "      <td>533</td>\n",
       "      <td>Q/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00387846</td>\n",
       "      <td>0.00183635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5455</th>\n",
       "      <td>22:40418743:G:GGCGGGT</td>\n",
       "      <td>22</td>\n",
       "      <td>40418743</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCGGGT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000196588</td>\n",
       "      <td>ENST00000355630</td>\n",
       "      <td>ENSP00000347847</td>\n",
       "      <td>MRTFA</td>\n",
       "      <td>2</td>\n",
       "      <td>665</td>\n",
       "      <td>A/APA</td>\n",
       "      <td>0.00285875</td>\n",
       "      <td>0.00383206</td>\n",
       "      <td>0.00471071</td>\n",
       "      <td>0.00488114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5456</th>\n",
       "      <td>22:46738126:T:TCCGCCGCCC</td>\n",
       "      <td>22</td>\n",
       "      <td>46738126</td>\n",
       "      <td>T</td>\n",
       "      <td>TCCGCCGCCC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000100422</td>\n",
       "      <td>ENST00000216264</td>\n",
       "      <td>ENSP00000216264</td>\n",
       "      <td>CERK</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>E/GAAE</td>\n",
       "      <td>0.000181475</td>\n",
       "      <td>0.003125</td>\n",
       "      <td>0.00222386</td>\n",
       "      <td>0.00403232</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5457 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            ID CHROM        POS   REF         ALT  \\\n",
       "0           10:97611611:TGAG:T    10   97611611  TGAG           T   \n",
       "1          10:124405531:TAGC:T    10  124405531  TAGC           T   \n",
       "2           19:33387464:GTCC:G    19   33387464  GTCC           G   \n",
       "3           22:42619859:ACAT:A    22   42619859  ACAT           A   \n",
       "4           22:42619781:CGAA:C    22   42619781  CGAA           C   \n",
       "...                        ...   ...        ...   ...         ...   \n",
       "5452     21:39779227:C:CGCTGCT    21   39779227     C     CGCTGCT   \n",
       "5453  22:20566526:A:ACAGCAGCAG    22   20566526     A  ACAGCAGCAG   \n",
       "5454     22:27798945:T:TTGCTGC    22   27798945     T     TTGCTGC   \n",
       "5455     22:40418743:G:GGCGGGT    22   40418743     G     GGCGGGT   \n",
       "5456  22:46738126:T:TCCGCCGCCC    22   46738126     T  TCCGCCGCCC   \n",
       "\n",
       "      Pathogenicity  indel             Gene          Feature  \\\n",
       "0        Pathogenic      3  ENSG00000241935  ENST00000370646   \n",
       "1        Pathogenic      3  ENSG00000065154  ENST00000368845   \n",
       "2        Pathogenic      3  ENSG00000124299  ENST00000244137   \n",
       "3        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "4        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "...             ...    ...              ...              ...   \n",
       "5452  gnomAD_Benign     -6  ENSG00000183067  ENST00000380588   \n",
       "5453  gnomAD_Benign     -9  ENSG00000099917  ENST00000263205   \n",
       "5454  gnomAD_Benign     -6  ENSG00000169184  ENST00000302326   \n",
       "5455  gnomAD_Benign     -6  ENSG00000196588  ENST00000355630   \n",
       "5456  gnomAD_Benign     -9  ENSG00000100422  ENST00000216264   \n",
       "\n",
       "     Protein.stable.ID  Symbol  Func Protein_position Amino_acids  \\\n",
       "0      ENSP00000359680   HOGA1     2              313         E/-   \n",
       "1      ENSP00000357838     OAT     2              184         A/-   \n",
       "2      ENSP00000244137    PEPD     2          453-454        ED/D   \n",
       "3      ENSP00000338461  CYB5R3     2              273         M/-   \n",
       "4      ENSP00000338461  CYB5R3     2              299         F/-   \n",
       "...                ...     ...   ...              ...         ...   \n",
       "5452   ENSP00000369962   IGSF5     2              286       R/RCC   \n",
       "5453   ENSP00000263205   MED15     2          250-251       -/QQQ   \n",
       "5454   ENSP00000304956     MN1     2              533       Q/QQQ   \n",
       "5455   ENSP00000347847   MRTFA     2              665       A/APA   \n",
       "5456   ENSP00000216264    CERK     2                8      E/GAAE   \n",
       "\n",
       "           gnomAD gnomAD_popmax      gnomAD3         UKBB  \n",
       "0     0.000465753   0.000123235  0.000269343  9.71885e-05  \n",
       "1               .             .            .            .  \n",
       "2     8.04531e-06   8.88952e-06  1.31416e-05  7.47604e-06  \n",
       "3               .             .            .            .  \n",
       "4               .             .            .            .  \n",
       "...           ...           ...          ...          ...  \n",
       "5452   0.00241128    0.00413725   0.00279994   0.00421913  \n",
       "5453            .             0    0.0022583   0.00060243  \n",
       "5454            .             0   0.00387846   0.00183635  \n",
       "5455   0.00285875    0.00383206   0.00471071   0.00488114  \n",
       "5456  0.000181475      0.003125   0.00222386   0.00403232  \n",
       "\n",
       "[5457 rows x 18 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "originaldata = pd.read_csv(\"../Data/Indel_curated_newmutseq.csv\")\n",
    "shine_del_1aa = pd.read_csv(\"../Data/SHINE/clinvar_gnomAD_del_1aa.txt\", sep = '\\t')\n",
    "shine_del_2aa = pd.read_csv(\"../Data/SHINE/clinvar_gnomAD_del_2aa.txt\", sep = '\\t')\n",
    "shine_ins_1aa = pd.read_csv(\"../Data/SHINE/clinvar_gnomAD_ins_1aa.txt\", sep = '\\t')\n",
    "shine_ins_2aa = pd.read_csv(\"../Data/SHINE/clinvar_gnomAD_ins_2aa.txt\", sep = '\\t')\n",
    "\n",
    "shine = pd.concat([shine_del_1aa, shine_del_2aa, shine_ins_1aa, shine_ins_2aa], ignore_index=True)\n",
    "len(shine) \n",
    "shine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Pathogenicity</th>\n",
       "      <th>indel</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Protein.stable.ID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Func</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD_popmax</th>\n",
       "      <th>gnomAD3</th>\n",
       "      <th>UKBB</th>\n",
       "      <th>wtseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:97611611:TGAG:T</td>\n",
       "      <td>10</td>\n",
       "      <td>97611611</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000241935</td>\n",
       "      <td>ENST00000370646</td>\n",
       "      <td>ENSP00000359680</td>\n",
       "      <td>HOGA1</td>\n",
       "      <td>2</td>\n",
       "      <td>313</td>\n",
       "      <td>E/-</td>\n",
       "      <td>0.000465753</td>\n",
       "      <td>0.000123235</td>\n",
       "      <td>0.000269343</td>\n",
       "      <td>9.71885e-05</td>\n",
       "      <td>MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:124405531:TAGC:T</td>\n",
       "      <td>10</td>\n",
       "      <td>124405531</td>\n",
       "      <td>TAGC</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000065154</td>\n",
       "      <td>ENST00000368845</td>\n",
       "      <td>ENSP00000357838</td>\n",
       "      <td>OAT</td>\n",
       "      <td>2</td>\n",
       "      <td>184</td>\n",
       "      <td>A/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19:33387464:GTCC:G</td>\n",
       "      <td>19</td>\n",
       "      <td>33387464</td>\n",
       "      <td>GTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000124299</td>\n",
       "      <td>ENST00000244137</td>\n",
       "      <td>ENSP00000244137</td>\n",
       "      <td>PEPD</td>\n",
       "      <td>2</td>\n",
       "      <td>453-454</td>\n",
       "      <td>ED/D</td>\n",
       "      <td>8.04531e-06</td>\n",
       "      <td>8.88952e-06</td>\n",
       "      <td>1.31416e-05</td>\n",
       "      <td>7.47604e-06</td>\n",
       "      <td>MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22:42619859:ACAT:A</td>\n",
       "      <td>22</td>\n",
       "      <td>42619859</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>M/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22:42619781:CGAA:C</td>\n",
       "      <td>22</td>\n",
       "      <td>42619781</td>\n",
       "      <td>CGAA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>299</td>\n",
       "      <td>F/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5452</th>\n",
       "      <td>21:39779227:C:CGCTGCT</td>\n",
       "      <td>21</td>\n",
       "      <td>39779227</td>\n",
       "      <td>C</td>\n",
       "      <td>CGCTGCT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000183067</td>\n",
       "      <td>ENST00000380588</td>\n",
       "      <td>ENSP00000369962</td>\n",
       "      <td>IGSF5</td>\n",
       "      <td>2</td>\n",
       "      <td>286</td>\n",
       "      <td>R/RCC</td>\n",
       "      <td>0.00241128</td>\n",
       "      <td>0.00413725</td>\n",
       "      <td>0.00279994</td>\n",
       "      <td>0.00421913</td>\n",
       "      <td>MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5453</th>\n",
       "      <td>22:20566526:A:ACAGCAGCAG</td>\n",
       "      <td>22</td>\n",
       "      <td>20566526</td>\n",
       "      <td>A</td>\n",
       "      <td>ACAGCAGCAG</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000099917</td>\n",
       "      <td>ENST00000263205</td>\n",
       "      <td>ENSP00000263205</td>\n",
       "      <td>MED15</td>\n",
       "      <td>2</td>\n",
       "      <td>250-251</td>\n",
       "      <td>-/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0022583</td>\n",
       "      <td>0.00060243</td>\n",
       "      <td>MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5454</th>\n",
       "      <td>22:27798945:T:TTGCTGC</td>\n",
       "      <td>22</td>\n",
       "      <td>27798945</td>\n",
       "      <td>T</td>\n",
       "      <td>TTGCTGC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000169184</td>\n",
       "      <td>ENST00000302326</td>\n",
       "      <td>ENSP00000304956</td>\n",
       "      <td>MN1</td>\n",
       "      <td>2</td>\n",
       "      <td>533</td>\n",
       "      <td>Q/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00387846</td>\n",
       "      <td>0.00183635</td>\n",
       "      <td>MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5455</th>\n",
       "      <td>22:40418743:G:GGCGGGT</td>\n",
       "      <td>22</td>\n",
       "      <td>40418743</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCGGGT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000196588</td>\n",
       "      <td>ENST00000355630</td>\n",
       "      <td>ENSP00000347847</td>\n",
       "      <td>MRTFA</td>\n",
       "      <td>2</td>\n",
       "      <td>665</td>\n",
       "      <td>A/APA</td>\n",
       "      <td>0.00285875</td>\n",
       "      <td>0.00383206</td>\n",
       "      <td>0.00471071</td>\n",
       "      <td>0.00488114</td>\n",
       "      <td>MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5456</th>\n",
       "      <td>22:46738126:T:TCCGCCGCCC</td>\n",
       "      <td>22</td>\n",
       "      <td>46738126</td>\n",
       "      <td>T</td>\n",
       "      <td>TCCGCCGCCC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000100422</td>\n",
       "      <td>ENST00000216264</td>\n",
       "      <td>ENSP00000216264</td>\n",
       "      <td>CERK</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>E/GAAE</td>\n",
       "      <td>0.000181475</td>\n",
       "      <td>0.003125</td>\n",
       "      <td>0.00222386</td>\n",
       "      <td>0.00403232</td>\n",
       "      <td>MGATGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAGAPG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5457 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            ID CHROM        POS   REF         ALT  \\\n",
       "0           10:97611611:TGAG:T    10   97611611  TGAG           T   \n",
       "1          10:124405531:TAGC:T    10  124405531  TAGC           T   \n",
       "2           19:33387464:GTCC:G    19   33387464  GTCC           G   \n",
       "3           22:42619859:ACAT:A    22   42619859  ACAT           A   \n",
       "4           22:42619781:CGAA:C    22   42619781  CGAA           C   \n",
       "...                        ...   ...        ...   ...         ...   \n",
       "5452     21:39779227:C:CGCTGCT    21   39779227     C     CGCTGCT   \n",
       "5453  22:20566526:A:ACAGCAGCAG    22   20566526     A  ACAGCAGCAG   \n",
       "5454     22:27798945:T:TTGCTGC    22   27798945     T     TTGCTGC   \n",
       "5455     22:40418743:G:GGCGGGT    22   40418743     G     GGCGGGT   \n",
       "5456  22:46738126:T:TCCGCCGCCC    22   46738126     T  TCCGCCGCCC   \n",
       "\n",
       "      Pathogenicity  indel             Gene          Feature  \\\n",
       "0        Pathogenic      3  ENSG00000241935  ENST00000370646   \n",
       "1        Pathogenic      3  ENSG00000065154  ENST00000368845   \n",
       "2        Pathogenic      3  ENSG00000124299  ENST00000244137   \n",
       "3        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "4        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "...             ...    ...              ...              ...   \n",
       "5452  gnomAD_Benign     -6  ENSG00000183067  ENST00000380588   \n",
       "5453  gnomAD_Benign     -9  ENSG00000099917  ENST00000263205   \n",
       "5454  gnomAD_Benign     -6  ENSG00000169184  ENST00000302326   \n",
       "5455  gnomAD_Benign     -6  ENSG00000196588  ENST00000355630   \n",
       "5456  gnomAD_Benign     -9  ENSG00000100422  ENST00000216264   \n",
       "\n",
       "     Protein.stable.ID  Symbol  Func Protein_position Amino_acids  \\\n",
       "0      ENSP00000359680   HOGA1     2              313         E/-   \n",
       "1      ENSP00000357838     OAT     2              184         A/-   \n",
       "2      ENSP00000244137    PEPD     2          453-454        ED/D   \n",
       "3      ENSP00000338461  CYB5R3     2              273         M/-   \n",
       "4      ENSP00000338461  CYB5R3     2              299         F/-   \n",
       "...                ...     ...   ...              ...         ...   \n",
       "5452   ENSP00000369962   IGSF5     2              286       R/RCC   \n",
       "5453   ENSP00000263205   MED15     2          250-251       -/QQQ   \n",
       "5454   ENSP00000304956     MN1     2              533       Q/QQQ   \n",
       "5455   ENSP00000347847   MRTFA     2              665       A/APA   \n",
       "5456   ENSP00000216264    CERK     2                8      E/GAAE   \n",
       "\n",
       "           gnomAD gnomAD_popmax      gnomAD3         UKBB  \\\n",
       "0     0.000465753   0.000123235  0.000269343  9.71885e-05   \n",
       "1               .             .            .            .   \n",
       "2     8.04531e-06   8.88952e-06  1.31416e-05  7.47604e-06   \n",
       "3               .             .            .            .   \n",
       "4               .             .            .            .   \n",
       "...           ...           ...          ...          ...   \n",
       "5452   0.00241128    0.00413725   0.00279994   0.00421913   \n",
       "5453            .             0    0.0022583   0.00060243   \n",
       "5454            .             0   0.00387846   0.00183635   \n",
       "5455   0.00285875    0.00383206   0.00471071   0.00488114   \n",
       "5456  0.000181475      0.003125   0.00222386   0.00403232   \n",
       "\n",
       "                                                  wtseq  \n",
       "0     MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...  \n",
       "1     MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...  \n",
       "2     MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...  \n",
       "3     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...  \n",
       "4     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...  \n",
       "...                                                 ...  \n",
       "5452  MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...  \n",
       "5453  MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...  \n",
       "5454  MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...  \n",
       "5455  MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...  \n",
       "5456  MGATGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAGAPG...  \n",
       "\n",
       "[5457 rows x 19 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Extracting protein IDs to fetch the protein seqeunces from BioMart (one seqeunce per ID)\n",
    "ProteinIDs_shine = shine['Protein.stable.ID'].drop_duplicates()\n",
    "ProteinIDs_shine.to_csv('ProteinIDs_shine.csv', index=False)\n",
    "\n",
    "protein_ids = []\n",
    "protein_seqs = []\n",
    "with open('../Data/SHINE/proteinIDs.fasta', 'r') as file:\n",
    "    current_id = None\n",
    "    current_seq = []\n",
    "\n",
    "    for line in file:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):  # If the line starts with '>', it's a new protein ID\n",
    "            if current_id is not None:\n",
    "                protein_ids.append(current_id)\n",
    "                protein_seqs.append(''.join(current_seq))  \n",
    "            current_id = line[1:]  \n",
    "            current_seq = []\n",
    "        else:\n",
    "            current_seq.append(line)\n",
    "\n",
    "    if current_id is not None:\n",
    "        protein_ids.append(current_id)\n",
    "        protein_seqs.append(''.join(current_seq))\n",
    "\n",
    "shine_seq = pd.DataFrame({\n",
    "    'ProteinID': protein_ids,\n",
    "    'Sequence': protein_seqs\n",
    "})\n",
    "\n",
    "\n",
    "#check that I have all the sequences\n",
    "len(ProteinIDs_shine), len(shine_seq['ProteinID']) #Doubled checked the missing IDs, not identified in BioMart, therefore these sequences have been updated and no longer aviliable. Revome them from shine dataset. \n",
    "\n",
    "\n",
    "shine = shine.merge(shine_seq[['ProteinID', 'Sequence']], left_on='Protein.stable.ID', right_on='ProteinID', how='left')\n",
    "shine.rename(columns={'Sequence': 'wtseq'}, inplace=True)\n",
    "shine.drop(columns=['ProteinID'], inplace=True)\n",
    "shine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Pathogenicity</th>\n",
       "      <th>indel</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Protein.stable.ID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Func</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD_popmax</th>\n",
       "      <th>gnomAD3</th>\n",
       "      <th>UKBB</th>\n",
       "      <th>wtseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:97611611:TGAG:T</td>\n",
       "      <td>10</td>\n",
       "      <td>97611611</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000241935</td>\n",
       "      <td>ENST00000370646</td>\n",
       "      <td>ENSP00000359680</td>\n",
       "      <td>HOGA1</td>\n",
       "      <td>2</td>\n",
       "      <td>313</td>\n",
       "      <td>E/-</td>\n",
       "      <td>0.000465753</td>\n",
       "      <td>0.000123235</td>\n",
       "      <td>0.000269343</td>\n",
       "      <td>9.71885e-05</td>\n",
       "      <td>MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:124405531:TAGC:T</td>\n",
       "      <td>10</td>\n",
       "      <td>124405531</td>\n",
       "      <td>TAGC</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000065154</td>\n",
       "      <td>ENST00000368845</td>\n",
       "      <td>ENSP00000357838</td>\n",
       "      <td>OAT</td>\n",
       "      <td>2</td>\n",
       "      <td>184</td>\n",
       "      <td>A/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19:33387464:GTCC:G</td>\n",
       "      <td>19</td>\n",
       "      <td>33387464</td>\n",
       "      <td>GTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000124299</td>\n",
       "      <td>ENST00000244137</td>\n",
       "      <td>ENSP00000244137</td>\n",
       "      <td>PEPD</td>\n",
       "      <td>2</td>\n",
       "      <td>453-454</td>\n",
       "      <td>ED/D</td>\n",
       "      <td>8.04531e-06</td>\n",
       "      <td>8.88952e-06</td>\n",
       "      <td>1.31416e-05</td>\n",
       "      <td>7.47604e-06</td>\n",
       "      <td>MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22:42619859:ACAT:A</td>\n",
       "      <td>22</td>\n",
       "      <td>42619859</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>M/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22:42619781:CGAA:C</td>\n",
       "      <td>22</td>\n",
       "      <td>42619781</td>\n",
       "      <td>CGAA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>CYB5R3</td>\n",
       "      <td>2</td>\n",
       "      <td>299</td>\n",
       "      <td>F/-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5452</th>\n",
       "      <td>21:39779227:C:CGCTGCT</td>\n",
       "      <td>21</td>\n",
       "      <td>39779227</td>\n",
       "      <td>C</td>\n",
       "      <td>CGCTGCT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000183067</td>\n",
       "      <td>ENST00000380588</td>\n",
       "      <td>ENSP00000369962</td>\n",
       "      <td>IGSF5</td>\n",
       "      <td>2</td>\n",
       "      <td>286</td>\n",
       "      <td>R/RCC</td>\n",
       "      <td>0.00241128</td>\n",
       "      <td>0.00413725</td>\n",
       "      <td>0.00279994</td>\n",
       "      <td>0.00421913</td>\n",
       "      <td>MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5453</th>\n",
       "      <td>22:20566526:A:ACAGCAGCAG</td>\n",
       "      <td>22</td>\n",
       "      <td>20566526</td>\n",
       "      <td>A</td>\n",
       "      <td>ACAGCAGCAG</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000099917</td>\n",
       "      <td>ENST00000263205</td>\n",
       "      <td>ENSP00000263205</td>\n",
       "      <td>MED15</td>\n",
       "      <td>2</td>\n",
       "      <td>250-251</td>\n",
       "      <td>-/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0022583</td>\n",
       "      <td>0.00060243</td>\n",
       "      <td>MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5454</th>\n",
       "      <td>22:27798945:T:TTGCTGC</td>\n",
       "      <td>22</td>\n",
       "      <td>27798945</td>\n",
       "      <td>T</td>\n",
       "      <td>TTGCTGC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000169184</td>\n",
       "      <td>ENST00000302326</td>\n",
       "      <td>ENSP00000304956</td>\n",
       "      <td>MN1</td>\n",
       "      <td>2</td>\n",
       "      <td>533</td>\n",
       "      <td>Q/QQQ</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00387846</td>\n",
       "      <td>0.00183635</td>\n",
       "      <td>MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5455</th>\n",
       "      <td>22:40418743:G:GGCGGGT</td>\n",
       "      <td>22</td>\n",
       "      <td>40418743</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCGGGT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000196588</td>\n",
       "      <td>ENST00000355630</td>\n",
       "      <td>ENSP00000347847</td>\n",
       "      <td>MRTFA</td>\n",
       "      <td>2</td>\n",
       "      <td>665</td>\n",
       "      <td>A/APA</td>\n",
       "      <td>0.00285875</td>\n",
       "      <td>0.00383206</td>\n",
       "      <td>0.00471071</td>\n",
       "      <td>0.00488114</td>\n",
       "      <td>MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5456</th>\n",
       "      <td>22:46738126:T:TCCGCCGCCC</td>\n",
       "      <td>22</td>\n",
       "      <td>46738126</td>\n",
       "      <td>T</td>\n",
       "      <td>TCCGCCGCCC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000100422</td>\n",
       "      <td>ENST00000216264</td>\n",
       "      <td>ENSP00000216264</td>\n",
       "      <td>CERK</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>E/GAAE</td>\n",
       "      <td>0.000181475</td>\n",
       "      <td>0.003125</td>\n",
       "      <td>0.00222386</td>\n",
       "      <td>0.00403232</td>\n",
       "      <td>MGATGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAGAPG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5452 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            ID CHROM        POS   REF         ALT  \\\n",
       "0           10:97611611:TGAG:T    10   97611611  TGAG           T   \n",
       "1          10:124405531:TAGC:T    10  124405531  TAGC           T   \n",
       "2           19:33387464:GTCC:G    19   33387464  GTCC           G   \n",
       "3           22:42619859:ACAT:A    22   42619859  ACAT           A   \n",
       "4           22:42619781:CGAA:C    22   42619781  CGAA           C   \n",
       "...                        ...   ...        ...   ...         ...   \n",
       "5452     21:39779227:C:CGCTGCT    21   39779227     C     CGCTGCT   \n",
       "5453  22:20566526:A:ACAGCAGCAG    22   20566526     A  ACAGCAGCAG   \n",
       "5454     22:27798945:T:TTGCTGC    22   27798945     T     TTGCTGC   \n",
       "5455     22:40418743:G:GGCGGGT    22   40418743     G     GGCGGGT   \n",
       "5456  22:46738126:T:TCCGCCGCCC    22   46738126     T  TCCGCCGCCC   \n",
       "\n",
       "      Pathogenicity  indel             Gene          Feature  \\\n",
       "0        Pathogenic      3  ENSG00000241935  ENST00000370646   \n",
       "1        Pathogenic      3  ENSG00000065154  ENST00000368845   \n",
       "2        Pathogenic      3  ENSG00000124299  ENST00000244137   \n",
       "3        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "4        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "...             ...    ...              ...              ...   \n",
       "5452  gnomAD_Benign     -6  ENSG00000183067  ENST00000380588   \n",
       "5453  gnomAD_Benign     -9  ENSG00000099917  ENST00000263205   \n",
       "5454  gnomAD_Benign     -6  ENSG00000169184  ENST00000302326   \n",
       "5455  gnomAD_Benign     -6  ENSG00000196588  ENST00000355630   \n",
       "5456  gnomAD_Benign     -9  ENSG00000100422  ENST00000216264   \n",
       "\n",
       "     Protein.stable.ID  Symbol  Func Protein_position Amino_acids  \\\n",
       "0      ENSP00000359680   HOGA1     2              313         E/-   \n",
       "1      ENSP00000357838     OAT     2              184         A/-   \n",
       "2      ENSP00000244137    PEPD     2          453-454        ED/D   \n",
       "3      ENSP00000338461  CYB5R3     2              273         M/-   \n",
       "4      ENSP00000338461  CYB5R3     2              299         F/-   \n",
       "...                ...     ...   ...              ...         ...   \n",
       "5452   ENSP00000369962   IGSF5     2              286       R/RCC   \n",
       "5453   ENSP00000263205   MED15     2          250-251       -/QQQ   \n",
       "5454   ENSP00000304956     MN1     2              533       Q/QQQ   \n",
       "5455   ENSP00000347847   MRTFA     2              665       A/APA   \n",
       "5456   ENSP00000216264    CERK     2                8      E/GAAE   \n",
       "\n",
       "           gnomAD gnomAD_popmax      gnomAD3         UKBB  \\\n",
       "0     0.000465753   0.000123235  0.000269343  9.71885e-05   \n",
       "1               .             .            .            .   \n",
       "2     8.04531e-06   8.88952e-06  1.31416e-05  7.47604e-06   \n",
       "3               .             .            .            .   \n",
       "4               .             .            .            .   \n",
       "...           ...           ...          ...          ...   \n",
       "5452   0.00241128    0.00413725   0.00279994   0.00421913   \n",
       "5453            .             0    0.0022583   0.00060243   \n",
       "5454            .             0   0.00387846   0.00183635   \n",
       "5455   0.00285875    0.00383206   0.00471071   0.00488114   \n",
       "5456  0.000181475      0.003125   0.00222386   0.00403232   \n",
       "\n",
       "                                                  wtseq  \n",
       "0     MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...  \n",
       "1     MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...  \n",
       "2     MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...  \n",
       "3     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...  \n",
       "4     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...  \n",
       "...                                                 ...  \n",
       "5452  MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...  \n",
       "5453  MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...  \n",
       "5454  MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...  \n",
       "5455  MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...  \n",
       "5456  MGATGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAGAPG...  \n",
       "\n",
       "[5452 rows x 19 columns]"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Remove indels with empty wtseq\n",
    "shine = shine.dropna(subset=['wtseq'])\n",
    "shine #dropped 6 indels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5452\n",
      "5451\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1609900/911969652.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['Protein_start'] = shine['Protein_position'].astype(str).apply(lambda x: x.split('-')[0])\n",
      "/tmp/ipykernel_1609900/911969652.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['Protein_start'] = shine['Protein_start'].astype(int)\n",
      "/tmp/ipykernel_1609900/911969652.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['wt'] = shine['Amino_acids'].astype(str).apply(lambda x: x.split('/')[0])\n",
      "/tmp/ipykernel_1609900/911969652.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['mut'] = shine['Amino_acids'].astype(str).apply(lambda x: x.split('/')[1] if '/' in x else '')\n",
      "/tmp/ipykernel_1609900/911969652.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['wt'] = shine['wt'].replace('-', '')\n",
      "/tmp/ipykernel_1609900/911969652.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  shine['mut'] = shine['mut'].replace('-', '')\n",
      "/tmp/ipykernel_1609900/911969652.py:19: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  shine = shine[~filter2]\n"
     ]
    }
   ],
   "source": [
    "#make mut sequences\n",
    "shine['Protein_start'] = shine['Protein_position'].astype(str).apply(lambda x: x.split('-')[0])\n",
    "shine['Protein_start'] = shine['Protein_start'].astype(int)\n",
    "shine['wt'] = shine['Amino_acids'].astype(str).apply(lambda x: x.split('/')[0])\n",
    "shine['mut'] = shine['Amino_acids'].astype(str).apply(lambda x: x.split('/')[1] if '/' in x else '')\n",
    "shine['wt'] = shine['wt'].replace('-', '')\n",
    "shine['mut'] = shine['mut'].replace('-', '')\n",
    "\n",
    "shine[['indel', 'Protein_position', 'Protein_start', 'Amino_acids', 'wt', 'mut']]\n",
    "\n",
    "print(len(shine))\n",
    "\n",
    "# Check if there is * or ? in the data\n",
    "filter = shine['wt'].str.contains(r'\\*|\\?', regex=True, na=False) \n",
    "filter2 = shine['mut'].str.contains(r'\\*|\\?', regex=True, na=False)\n",
    "\n",
    "# Drops rows with * or ?\n",
    "shine = shine[~filter]\n",
    "shine = shine[~filter2]\n",
    "\n",
    "print(len(shine)) #removed 1 row\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mutate(sequence, protein_start, wt, mut):\n",
    "    protein_start = int(protein_start) - 1\n",
    "    wt = str(wt)\n",
    "    mut = str(mut)\n",
    "    wt_len = len(wt)\n",
    "    sequence_before_wt = sequence[:protein_start]\n",
    "    sequence_after_wt = sequence[protein_start + wt_len:]\n",
    "    return sequence_before_wt + mut + sequence_after_wt\n",
    "\n",
    "# Apply the function to create mutated sequences\n",
    "shine['mutseq'] = shine.apply(lambda row: mutate(row['wtseq'], row['Protein_start'], row['wt'], row['mut']), axis=1)\n",
    "\n",
    "# Add information on indel type and length\n",
    "shine['length_wt'] = shine['wtseq'].apply(len)\n",
    "shine['length_mut'] = shine['mutseq'].apply(len)\n",
    "shine['length_indel'] = shine.apply(lambda row: abs(row['length_mut'] - row['length_wt']), axis=1)\n",
    "shine['indel_type'] = shine.apply(lambda row: 'insertion' if row['length_mut'] > row['length_wt'] else 'deletion', axis=1)\n",
    "shine['Consequence'] = shine.apply(lambda row: 'insertion' if row['indel'] < 0 else 'deletion', axis=1)\n",
    "\n",
    "shine['check'] = shine.apply(lambda row: row['indel_type'] in row['Consequence'], axis=1)\n",
    "shine['check'].value_counts() #one sequence is false due to the wtseq not long enough to insert the mut. Removed this indel\n",
    "shine[shine['length_indel'] == 0] #one sequence is 0 due to the wtseq not long enough to insert the mut. Removed this indel\n",
    "\n",
    "shine = shine[shine['check'] == True]\n",
    "shine = shine[shine['length_indel'] != 0]\n",
    "len(shine)\n",
    "shine.to_csv(\"Indel_shine.csv\", sep=',')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Id_matching</th>\n",
       "      <th>Feature_x</th>\n",
       "      <th>Consequence_x</th>\n",
       "      <th>Protein_start</th>\n",
       "      <th>Protein_stop</th>\n",
       "      <th>wt</th>\n",
       "      <th>mut</th>\n",
       "      <th>seq</th>\n",
       "      <th>...</th>\n",
       "      <th>PROVEAN</th>\n",
       "      <th>SIFT-indel</th>\n",
       "      <th>VEST-indel</th>\n",
       "      <th>VVP</th>\n",
       "      <th>length_wt</th>\n",
       "      <th>length_mut</th>\n",
       "      <th>length_indel</th>\n",
       "      <th>indel_type</th>\n",
       "      <th>check</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000379198.5</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>LRRAWR</td>\n",
       "      <td>MKLLRRAWRRRAALGLGTLALCGAALLYLARCAAEPGDPRAMSGRS...</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.44</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.05</td>\n",
       "      <td>7.0</td>\n",
       "      <td>330</td>\n",
       "      <td>335</td>\n",
       "      <td>5</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "      <td>insertion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>ENST00000379198.5</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>136</td>\n",
       "      <td>139.0</td>\n",
       "      <td>VLAM</td>\n",
       "      <td>V</td>\n",
       "      <td>MKLLRRAWRRRAALGLGTLALCGAALLYLARCAAEPGDPRAMSGRS...</td>\n",
       "      <td>...</td>\n",
       "      <td>-11.00</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.23</td>\n",
       "      <td>98.0</td>\n",
       "      <td>330</td>\n",
       "      <td>327</td>\n",
       "      <td>3</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>ENST00000476993.2</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>61</td>\n",
       "      <td>69.0</td>\n",
       "      <td>VAPLPLGTG</td>\n",
       "      <td>V</td>\n",
       "      <td>MLPWTALGLALSLRLALARSGAERGPPASAPRGDLMFLLDSSASVS...</td>\n",
       "      <td>...</td>\n",
       "      <td>-27.11</td>\n",
       "      <td>neutral</td>\n",
       "      <td>0.07</td>\n",
       "      <td>100.0</td>\n",
       "      <td>446</td>\n",
       "      <td>438</td>\n",
       "      <td>8</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>ENST00000404249.8</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>312</td>\n",
       "      <td>314.0</td>\n",
       "      <td>EEE</td>\n",
       "      <td>E</td>\n",
       "      <td>MGDEKDSWKVKTLDEILQEKKRRKEQEEKAEIKRLKNSDDRDSKRD...</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.76</td>\n",
       "      <td>neutral</td>\n",
       "      <td>0.10</td>\n",
       "      <td>15.0</td>\n",
       "      <td>781</td>\n",
       "      <td>779</td>\n",
       "      <td>2</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>ENST00000341426.9</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>445</td>\n",
       "      <td>446.0</td>\n",
       "      <td>EG</td>\n",
       "      <td>G</td>\n",
       "      <td>MEMEQEKMTMNKELSPDAAAYCCSACHGDETWSYNHPIRGRAKSRS...</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.90</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.29</td>\n",
       "      <td>95.0</td>\n",
       "      <td>447</td>\n",
       "      <td>446</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3812</th>\n",
       "      <td>3815</td>\n",
       "      <td>3815</td>\n",
       "      <td>3959</td>\n",
       "      <td>ENST00000369550.10</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>36</td>\n",
       "      <td>37.0</td>\n",
       "      <td>FL</td>\n",
       "      <td>F</td>\n",
       "      <td>MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...</td>\n",
       "      <td>...</td>\n",
       "      <td>-7.29</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.98</td>\n",
       "      <td>93.0</td>\n",
       "      <td>515</td>\n",
       "      <td>514</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3813</th>\n",
       "      <td>3816</td>\n",
       "      <td>3816</td>\n",
       "      <td>3960</td>\n",
       "      <td>ENST00000369550.10</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.20</td>\n",
       "      <td>neutral</td>\n",
       "      <td>0.37</td>\n",
       "      <td>87.0</td>\n",
       "      <td>515</td>\n",
       "      <td>514</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3814</th>\n",
       "      <td>3817</td>\n",
       "      <td>3817</td>\n",
       "      <td>3961</td>\n",
       "      <td>ENST00000369550.10</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>498</td>\n",
       "      <td>499.0</td>\n",
       "      <td>KK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.88</td>\n",
       "      <td>neutral</td>\n",
       "      <td>0.19</td>\n",
       "      <td>93.0</td>\n",
       "      <td>515</td>\n",
       "      <td>513</td>\n",
       "      <td>2</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3815</th>\n",
       "      <td>3818</td>\n",
       "      <td>3818</td>\n",
       "      <td>3962</td>\n",
       "      <td>ENST00000360256.9</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>2136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>AYRG</td>\n",
       "      <td>MQIELSTCFFLCLLRFCFSATRRYYLGAVELSWDYMQSDLGELPVD...</td>\n",
       "      <td>...</td>\n",
       "      <td>-10.11</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.93</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2352</td>\n",
       "      <td>2355</td>\n",
       "      <td>3</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "      <td>insertion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3816</th>\n",
       "      <td>3819</td>\n",
       "      <td>3819</td>\n",
       "      <td>3963</td>\n",
       "      <td>ENST00000360256.9</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>491</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>PRP</td>\n",
       "      <td>MQIELSTCFFLCLLRFCFSATRRYYLGAVELSWDYMQSDLGELPVD...</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.60</td>\n",
       "      <td>damaging</td>\n",
       "      <td>0.89</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2352</td>\n",
       "      <td>2354</td>\n",
       "      <td>2</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "      <td>insertion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3817 rows Ã— 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0.1  Unnamed: 0  Id_matching           Feature_x  \\\n",
       "0                0           0            0   ENST00000379198.5   \n",
       "1                1           1            1   ENST00000379198.5   \n",
       "2                2           2            2   ENST00000476993.2   \n",
       "3                3           3            3   ENST00000404249.8   \n",
       "4                4           4            4   ENST00000341426.9   \n",
       "...            ...         ...          ...                 ...   \n",
       "3812          3815        3815         3959  ENST00000369550.10   \n",
       "3813          3816        3816         3960  ENST00000369550.10   \n",
       "3814          3817        3817         3961  ENST00000369550.10   \n",
       "3815          3818        3818         3962   ENST00000360256.9   \n",
       "3816          3819        3819         3963   ENST00000360256.9   \n",
       "\n",
       "          Consequence_x  Protein_start  Protein_stop         wt     mut  \\\n",
       "0     inframe_insertion              4           NaN          L  LRRAWR   \n",
       "1      inframe_deletion            136         139.0       VLAM       V   \n",
       "2      inframe_deletion             61          69.0  VAPLPLGTG       V   \n",
       "3      inframe_deletion            312         314.0        EEE       E   \n",
       "4      inframe_deletion            445         446.0         EG       G   \n",
       "...                 ...            ...           ...        ...     ...   \n",
       "3812   inframe_deletion             36          37.0         FL       F   \n",
       "3813   inframe_deletion            470           NaN          K     NaN   \n",
       "3814   inframe_deletion            498         499.0         KK     NaN   \n",
       "3815  inframe_insertion           2136           NaN          G    AYRG   \n",
       "3816  inframe_insertion            491           NaN          P     PRP   \n",
       "\n",
       "                                                    seq  ... PROVEAN  \\\n",
       "0     MKLLRRAWRRRAALGLGTLALCGAALLYLARCAAEPGDPRAMSGRS...  ...   -3.44   \n",
       "1     MKLLRRAWRRRAALGLGTLALCGAALLYLARCAAEPGDPRAMSGRS...  ...  -11.00   \n",
       "2     MLPWTALGLALSLRLALARSGAERGPPASAPRGDLMFLLDSSASVS...  ...  -27.11   \n",
       "3     MGDEKDSWKVKTLDEILQEKKRRKEQEEKAEIKRLKNSDDRDSKRD...  ...   -2.76   \n",
       "4     MEMEQEKMTMNKELSPDAAAYCCSACHGDETWSYNHPIRGRAKSRS...  ...   -1.90   \n",
       "...                                                 ...  ...     ...   \n",
       "3812  MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...  ...   -7.29   \n",
       "3813  MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...  ...   -2.20   \n",
       "3814  MADAEVIILPKKHKKKKERKSLPEEDVAEIQHAEEFLIKPESKVAK...  ...   -4.88   \n",
       "3815  MQIELSTCFFLCLLRFCFSATRRYYLGAVELSWDYMQSDLGELPVD...  ...  -10.11   \n",
       "3816  MQIELSTCFFLCLLRFCFSATRRYYLGAVELSWDYMQSDLGELPVD...  ...   -9.60   \n",
       "\n",
       "     SIFT-indel VEST-indel    VVP length_wt  length_mut length_indel  \\\n",
       "0      damaging       0.05    7.0       330         335            5   \n",
       "1      damaging       0.23   98.0       330         327            3   \n",
       "2       neutral       0.07  100.0       446         438            8   \n",
       "3       neutral       0.10   15.0       781         779            2   \n",
       "4      damaging       0.29   95.0       447         446            1   \n",
       "...         ...        ...    ...       ...         ...          ...   \n",
       "3812   damaging       0.98   93.0       515         514            1   \n",
       "3813    neutral       0.37   87.0       515         514            1   \n",
       "3814    neutral       0.19   93.0       515         513            2   \n",
       "3815   damaging       0.93  100.0      2352        2355            3   \n",
       "3816   damaging       0.89  100.0      2352        2354            2   \n",
       "\n",
       "      indel_type  check Consequence  \n",
       "0      insertion   True   insertion  \n",
       "1       deletion   True    deletion  \n",
       "2       deletion   True    deletion  \n",
       "3       deletion   True    deletion  \n",
       "4       deletion   True    deletion  \n",
       "...          ...    ...         ...  \n",
       "3812    deletion   True    deletion  \n",
       "3813    deletion   True    deletion  \n",
       "3814    deletion   True    deletion  \n",
       "3815   insertion   True   insertion  \n",
       "3816   insertion   True   insertion  \n",
       "\n",
       "[3817 rows x 48 columns]"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Add the indel type columns to origional data also\n",
    "originaldata['length_wt'] = originaldata['seq'].apply(len)\n",
    "originaldata['length_mut'] = originaldata['mutseq'].apply(len)\n",
    "originaldata['length_indel'] = originaldata.apply(lambda row: abs(row['length_mut'] - row['length_wt']), axis=1)\n",
    "originaldata['indel_type'] = originaldata.apply(lambda row: 'insertion' if row['length_mut'] > row['length_wt'] else 'deletion', axis=1)\n",
    "originaldata['Consequence'] = originaldata.apply(lambda row: 'insertion' if row['Consequence_x'] == 'inframe_insertion' else 'deletion', axis=1)\n",
    "originaldata['check'] = originaldata.apply(lambda row: row['indel_type'] in row['Consequence'], axis=1)\n",
    "originaldata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Classification</th>\n",
       "      <th>indel</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Feature_x</th>\n",
       "      <th>Protein.stable.ID</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein_start</th>\n",
       "      <th>wt</th>\n",
       "      <th>mut</th>\n",
       "      <th>mutseq</th>\n",
       "      <th>length_wt</th>\n",
       "      <th>length_mut</th>\n",
       "      <th>length_indel</th>\n",
       "      <th>indel_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:97611611:TGAG:T</td>\n",
       "      <td>10</td>\n",
       "      <td>97611611</td>\n",
       "      <td>TGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000241935</td>\n",
       "      <td>ENST00000370646</td>\n",
       "      <td>ENSP00000359680</td>\n",
       "      <td>...</td>\n",
       "      <td>313</td>\n",
       "      <td>E</td>\n",
       "      <td></td>\n",
       "      <td>MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...</td>\n",
       "      <td>328</td>\n",
       "      <td>327</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:124405531:TAGC:T</td>\n",
       "      <td>10</td>\n",
       "      <td>124405531</td>\n",
       "      <td>TAGC</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000065154</td>\n",
       "      <td>ENST00000368845</td>\n",
       "      <td>ENSP00000357838</td>\n",
       "      <td>...</td>\n",
       "      <td>184</td>\n",
       "      <td>A</td>\n",
       "      <td></td>\n",
       "      <td>MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...</td>\n",
       "      <td>440</td>\n",
       "      <td>439</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19:33387464:GTCC:G</td>\n",
       "      <td>19</td>\n",
       "      <td>33387464</td>\n",
       "      <td>GTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000124299</td>\n",
       "      <td>ENST00000244137</td>\n",
       "      <td>ENSP00000244137</td>\n",
       "      <td>...</td>\n",
       "      <td>453</td>\n",
       "      <td>ED</td>\n",
       "      <td>D</td>\n",
       "      <td>MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...</td>\n",
       "      <td>494</td>\n",
       "      <td>493</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22:42619859:ACAT:A</td>\n",
       "      <td>22</td>\n",
       "      <td>42619859</td>\n",
       "      <td>ACAT</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>...</td>\n",
       "      <td>273</td>\n",
       "      <td>M</td>\n",
       "      <td></td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "      <td>302</td>\n",
       "      <td>301</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22:42619781:CGAA:C</td>\n",
       "      <td>22</td>\n",
       "      <td>42619781</td>\n",
       "      <td>CGAA</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000100243</td>\n",
       "      <td>ENST00000352397</td>\n",
       "      <td>ENSP00000338461</td>\n",
       "      <td>...</td>\n",
       "      <td>299</td>\n",
       "      <td>F</td>\n",
       "      <td></td>\n",
       "      <td>MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...</td>\n",
       "      <td>302</td>\n",
       "      <td>301</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion</td>\n",
       "      <td>deletion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5452</th>\n",
       "      <td>21:39779227:C:CGCTGCT</td>\n",
       "      <td>21</td>\n",
       "      <td>39779227</td>\n",
       "      <td>C</td>\n",
       "      <td>CGCTGCT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000183067</td>\n",
       "      <td>ENST00000380588</td>\n",
       "      <td>ENSP00000369962</td>\n",
       "      <td>...</td>\n",
       "      <td>286</td>\n",
       "      <td>R</td>\n",
       "      <td>RCC</td>\n",
       "      <td>MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...</td>\n",
       "      <td>408</td>\n",
       "      <td>410</td>\n",
       "      <td>2</td>\n",
       "      <td>insertion</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5453</th>\n",
       "      <td>22:20566526:A:ACAGCAGCAG</td>\n",
       "      <td>22</td>\n",
       "      <td>20566526</td>\n",
       "      <td>A</td>\n",
       "      <td>ACAGCAGCAG</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000099917</td>\n",
       "      <td>ENST00000263205</td>\n",
       "      <td>ENSP00000263205</td>\n",
       "      <td>...</td>\n",
       "      <td>250</td>\n",
       "      <td></td>\n",
       "      <td>QQQ</td>\n",
       "      <td>MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...</td>\n",
       "      <td>789</td>\n",
       "      <td>792</td>\n",
       "      <td>3</td>\n",
       "      <td>insertion</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5454</th>\n",
       "      <td>22:27798945:T:TTGCTGC</td>\n",
       "      <td>22</td>\n",
       "      <td>27798945</td>\n",
       "      <td>T</td>\n",
       "      <td>TTGCTGC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000169184</td>\n",
       "      <td>ENST00000302326</td>\n",
       "      <td>ENSP00000304956</td>\n",
       "      <td>...</td>\n",
       "      <td>533</td>\n",
       "      <td>Q</td>\n",
       "      <td>QQQ</td>\n",
       "      <td>MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...</td>\n",
       "      <td>1321</td>\n",
       "      <td>1323</td>\n",
       "      <td>2</td>\n",
       "      <td>insertion</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5455</th>\n",
       "      <td>22:40418743:G:GGCGGGT</td>\n",
       "      <td>22</td>\n",
       "      <td>40418743</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCGGGT</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-6</td>\n",
       "      <td>ENSG00000196588</td>\n",
       "      <td>ENST00000355630</td>\n",
       "      <td>ENSP00000347847</td>\n",
       "      <td>...</td>\n",
       "      <td>665</td>\n",
       "      <td>A</td>\n",
       "      <td>APA</td>\n",
       "      <td>MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...</td>\n",
       "      <td>1032</td>\n",
       "      <td>1034</td>\n",
       "      <td>2</td>\n",
       "      <td>insertion</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5456</th>\n",
       "      <td>22:46738126:T:TCCGCCGCCC</td>\n",
       "      <td>22</td>\n",
       "      <td>46738126</td>\n",
       "      <td>T</td>\n",
       "      <td>TCCGCCGCCC</td>\n",
       "      <td>gnomAD_Benign</td>\n",
       "      <td>-9</td>\n",
       "      <td>ENSG00000100422</td>\n",
       "      <td>ENST00000216264</td>\n",
       "      <td>ENSP00000216264</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>E</td>\n",
       "      <td>GAAE</td>\n",
       "      <td>MGATGAAGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAG...</td>\n",
       "      <td>538</td>\n",
       "      <td>541</td>\n",
       "      <td>3</td>\n",
       "      <td>insertion</td>\n",
       "      <td>insertion</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5449 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            ID CHROM        POS   REF         ALT  \\\n",
       "0           10:97611611:TGAG:T    10   97611611  TGAG           T   \n",
       "1          10:124405531:TAGC:T    10  124405531  TAGC           T   \n",
       "2           19:33387464:GTCC:G    19   33387464  GTCC           G   \n",
       "3           22:42619859:ACAT:A    22   42619859  ACAT           A   \n",
       "4           22:42619781:CGAA:C    22   42619781  CGAA           C   \n",
       "...                        ...   ...        ...   ...         ...   \n",
       "5452     21:39779227:C:CGCTGCT    21   39779227     C     CGCTGCT   \n",
       "5453  22:20566526:A:ACAGCAGCAG    22   20566526     A  ACAGCAGCAG   \n",
       "5454     22:27798945:T:TTGCTGC    22   27798945     T     TTGCTGC   \n",
       "5455     22:40418743:G:GGCGGGT    22   40418743     G     GGCGGGT   \n",
       "5456  22:46738126:T:TCCGCCGCCC    22   46738126     T  TCCGCCGCCC   \n",
       "\n",
       "     Classification  indel           GeneID        Feature_x  \\\n",
       "0        Pathogenic      3  ENSG00000241935  ENST00000370646   \n",
       "1        Pathogenic      3  ENSG00000065154  ENST00000368845   \n",
       "2        Pathogenic      3  ENSG00000124299  ENST00000244137   \n",
       "3        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "4        Pathogenic      3  ENSG00000100243  ENST00000352397   \n",
       "...             ...    ...              ...              ...   \n",
       "5452  gnomAD_Benign     -6  ENSG00000183067  ENST00000380588   \n",
       "5453  gnomAD_Benign     -9  ENSG00000099917  ENST00000263205   \n",
       "5454  gnomAD_Benign     -6  ENSG00000169184  ENST00000302326   \n",
       "5455  gnomAD_Benign     -6  ENSG00000196588  ENST00000355630   \n",
       "5456  gnomAD_Benign     -9  ENSG00000100422  ENST00000216264   \n",
       "\n",
       "     Protein.stable.ID  ... Protein_start  wt   mut  \\\n",
       "0      ENSP00000359680  ...           313   E         \n",
       "1      ENSP00000357838  ...           184   A         \n",
       "2      ENSP00000244137  ...           453  ED     D   \n",
       "3      ENSP00000338461  ...           273   M         \n",
       "4      ENSP00000338461  ...           299   F         \n",
       "...                ...  ...           ...  ..   ...   \n",
       "5452   ENSP00000369962  ...           286   R   RCC   \n",
       "5453   ENSP00000263205  ...           250       QQQ   \n",
       "5454   ENSP00000304956  ...           533   Q   QQQ   \n",
       "5455   ENSP00000347847  ...           665   A   APA   \n",
       "5456   ENSP00000216264  ...             8   E  GAAE   \n",
       "\n",
       "                                                 mutseq length_wt length_mut  \\\n",
       "0     MLGPQVWSSVRQGLSRSLSRNVGVWASGEGKKVDIAGIYPPVTTPF...       328        327   \n",
       "1     MFSKLAHLQRFAVLSRGVHSSVASATSVATKKTVQGPPTSDDIFER...       440        439   \n",
       "2     MAAATGPSFWLGNETLKVPLALFALNRQRLCERLRKNPAVQAGSIV...       494        493   \n",
       "3     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...       302        301   \n",
       "4     MGAQLSTLGHMVLFPVWFLYSLLMKLFQRSTPAITLESPDIKYPLR...       302        301   \n",
       "...                                                 ...       ...        ...   \n",
       "5452  MGQKERSTADTLPDLEEWKSAAGLRWWQTAVVDGSGSGNEVIEGPQ...       408        410   \n",
       "5453  MDVSGQETDWRSTAFRQKLVSQIEDAMRKAGVAHSKSSKDMESHVF...       789        792   \n",
       "5454  MFGLDQFEPQVNSRNAGQGERNFNETGLSMNTHFKAPAFHTGGPPG...      1321       1323   \n",
       "5455  MDFSSVVCLPPSVIAVNGLDGGGAGENDDEPVLVSLSAAPSPQSEA...      1032       1034   \n",
       "5456  MGATGAAGAAEPLQSVLWVKQQRCAVSLEPARALLRWWRSPGPGAG...       538        541   \n",
       "\n",
       "     length_indel indel_type Consequence  check  \n",
       "0               1   deletion    deletion   True  \n",
       "1               1   deletion    deletion   True  \n",
       "2               1   deletion    deletion   True  \n",
       "3               1   deletion    deletion   True  \n",
       "4               1   deletion    deletion   True  \n",
       "...           ...        ...         ...    ...  \n",
       "5452            2  insertion   insertion   True  \n",
       "5453            3  insertion   insertion   True  \n",
       "5454            2  insertion   insertion   True  \n",
       "5455            2  insertion   insertion   True  \n",
       "5456            3  insertion   insertion   True  \n",
       "\n",
       "[5449 rows x 29 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shine.rename(columns={'Feature': 'Feature_x'}, inplace=True)\n",
    "shine.rename(columns={'Gene': 'GeneID'}, inplace=True)\n",
    "shine.rename(columns={'Pathogenicity': 'Classification'}, inplace=True)\n",
    "shine.rename(columns={'Symbol': 'SYMBOL'}, inplace=True)\n",
    "shine.rename(columns={'wtseq': 'seq'}, inplace=True)\n",
    "shine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "length_indel\n",
       "False    6202\n",
       "True      264\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "overlaps = shine[shine['mutseq'].isin(originaldata['mutseq'])]\n",
    "overlaps\n",
    "\n",
    "shine_unique = shine[~shine['mutseq'].isin(overlaps['mutseq'])]\n",
    "shine\n",
    "\n",
    "\n",
    "\n",
    "indel_extended = pd.concat([originaldata, shine_unique])\n",
    "indel_extended['labels_cleaned'] = np.where(indel_extended['Classification'].str.contains('benign', case=False, na=False), \n",
    "                                'Benign', \n",
    "                                np.where(indel_extended['Classification'].str.contains('pathogenic', case=False, na=False), \n",
    "                                         'Pathogenic', \n",
    "                                         np.nan))\n",
    "indel_extended.to_csv('Indel_extended.csv')\n",
    "\n",
    "\n",
    "len(indel_extended)\n",
    "indel_extended['indel_type'].value_counts()\n",
    "indel_extended['labels_cleaned'].value_counts()\n",
    "(indel_extended['length_indel'] == 1).value_counts()\n",
    "\n",
    "group2to4 = indel_extended[(indel_extended['length_indel'] >= 2) & (indel_extended['length_indel'] <= 4)]\n",
    "len(group2to4)\n",
    "group5to10 = indel_extended[(indel_extended['length_indel'] >= 5) & (indel_extended['length_indel'] <= 10)]\n",
    "len(group5to10)\n",
    "(indel_extended['length_indel'] >= 11).value_counts()\n",
    "\n",
    "#Remove duplicates, after created identical sequences. \n",
    "#combine with our dataset (if we have unique sequences)\n",
    "#check the natural distribution of indels human genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "naturedata = pd.read_csv('ClinVar_indel_benchmark_with_predictions.csv')\n",
    "naturedata['full_alt_seq']\n",
    "\n",
    "naturedata['contains'] = naturedata['window_alt_seq'].apply(lambda x: any(indel_extended['mutseq'].str.contains(x, regex=False)))\n",
    "naturedata['contains'].value_counts()\n",
    "\n",
    "\n",
    "naturedata_unique = naturedata[naturedata['contains'] == False]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1609900/3731942326.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique['indel_type'] = naturedata_unique.apply(lambda row: 'insertion' if row['ref_alt_length_diff'] < 0 else 'deletion', axis=1)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "indel_type\n",
       "deletion     689\n",
       "insertion    414\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "naturedata_unique['mut_type'].value_counts()\n",
    "naturedata_unique['indel_type'] = naturedata_unique.apply(lambda row: 'insertion' if row['ref_alt_length_diff'] < 0 else 'deletion', axis=1)\n",
    "naturedata_unique['indel_type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1609900/2662015425.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'start_pos': 'Protein_start'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'end_pos': 'Protein_stop'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'site_ref_seq': 'wt'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'site_alt_seq': 'mut'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'window_ref_seq': 'seq'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'window_alt_seq': 'mutseq'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique.rename(columns={'num_aa_affected': 'length_indel'}, inplace=True)\n",
      "/tmp/ipykernel_1609900/2662015425.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  naturedata_unique['labels_cleaned'] = naturedata_unique.apply(lambda row: 'Pathogenic' if row['label'] == 1 else 'Benign', axis=1)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "length_indel\n",
       "1      378\n",
       "2      310\n",
       "3       95\n",
       "4       78\n",
       "6       60\n",
       "5       50\n",
       "7       29\n",
       "9       23\n",
       "8       23\n",
       "10      12\n",
       "13       5\n",
       "12       5\n",
       "11       5\n",
       "21       3\n",
       "14       3\n",
       "18       2\n",
       "32       2\n",
       "22       2\n",
       "20       2\n",
       "26       2\n",
       "16       2\n",
       "145      1\n",
       "24       1\n",
       "197      1\n",
       "58       1\n",
       "39       1\n",
       "207      1\n",
       "27       1\n",
       "15       1\n",
       "66       1\n",
       "204      1\n",
       "152      1\n",
       "59       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#cleanup the naturedata column names to match our indel_extended\n",
    "\n",
    "naturedata_unique.rename(columns={'start_pos': 'Protein_start'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'end_pos': 'Protein_stop'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'site_ref_seq': 'wt'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'site_alt_seq': 'mut'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'window_ref_seq': 'seq'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'window_alt_seq': 'mutseq'}, inplace=True)\n",
    "naturedata_unique.rename(columns={'num_aa_affected': 'length_indel'}, inplace=True)\n",
    "naturedata_unique['labels_cleaned'] = naturedata_unique.apply(lambda row: 'Pathogenic' if row['label'] == 1 else 'Benign', axis=1)\n",
    "\n",
    "naturedata_unique['length_indel'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7569"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel_extended = pd.concat([indel_extended, naturedata_unique])\n",
    "indel_extended.to_csv('Indel_extended.csv')\n",
    "len(indel_extended)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "indel_type\n",
       "deletion     5157\n",
       "insertion    2412\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel_extended['indel_type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "labels_cleaned\n",
       "Benign        4638\n",
       "Pathogenic    2931\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel_extended['labels_cleaned'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "length_indel\n",
       "False    3868\n",
       "True     3701\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(indel_extended['length_indel'] == 1).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2512"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group2to4 = indel_extended[(indel_extended['length_indel'] >= 2) & (indel_extended['length_indel'] <= 4)]\n",
    "len(group2to4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1047"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group5to10 = indel_extended[(indel_extended['length_indel'] >= 5) & (indel_extended['length_indel'] <= 10)]\n",
    "len(group5to10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "length_indel\n",
       "False    7260\n",
       "True      309\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(indel_extended['length_indel'] >= 11).value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Initial outlook on the SHINE indel dataset\n",
    "Finding the 1-2 aa indels and comparing to our dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [],
   "source": [
    "#shine['indel'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [],
   "source": [
    "#shine_1to2aa = shine[(shine['indel'] == 3) | (shine['indel'] == -3) | (shine['indel'] == 6) | (shine['indel'] == -6) | (shine['indel'] == 9) | (shine['indel'] == -9)]\n",
    "#len(shine_1to2aa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [],
   "source": [
    "#originaldata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "#shine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Check for overlapping indels in alldata\n",
    "#overlapping_genes = originaldata[originaldata['GeneID'].isin(shine['Gene'])] \n",
    "#overlapping_genes2 = shine[shine['Gene'].isin(originaldata['GeneID'])]\n",
    "#overlapping_genes2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Removing overlapping Genes that we allready have in the original data from the shine dataset\n",
    "#shine_unique = shine[~shine['Gene'].isin(overlapping_genes['GeneID'])]\n",
    "#our_unique = originaldata[~originaldata['GeneID'].isin(overlapping_genes2['Gene'])]\n",
    "#shine_unique_1to2aa = shine_1to2aa[~shine_1to2aa['Gene'].isin(overlapping_genes['GeneID'])]\n",
    "\n",
    "#len(shine), len(shine_1to2aa), len(shine_unique), len(our_unique), len(shine_unique_1to2aa)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "MDAnalysis",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
